programs/uxd/src/instructions/compute_depositories_targets.rs

```diff
@@ -0,0 +1,261 @@
+use crate::state::MercurialVaultDepository;
+use crate::utils::checked_convert_u128_to_u64;
+use crate::utils::compute_amount_fraction_floor;
+use crate::validate_is_program_frozen;
+use crate::Controller;
+use crate::BPS_UNIT_CONVERSION;
+use crate::CONTROLLER_NAMESPACE;
+use crate::CREDIX_LP_DEPOSITORY_NAMESPACE;
+use crate::MERCURIAL_VAULT_DEPOSITORY_NAMESPACE;
+use anchor_lang::prelude::*;
+
+#[derive(Accounts)]
+pub struct ComputeDepositoriesTargets<'info> {
+    /// #1 Permissionless IX that can be called by anyone
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    /// #2 The top level UXDProgram on chain account managing the redeemable mint
+    #[account(
+        mut,
+        seeds = [CONTROLLER_NAMESPACE],
+        bump = controller.load()?.bump,
+        constraint = controller.load()?.registered_mercurial_vault_depositories.contains(&mercurial_vault_depository.key()) @UxdError::InvalidDepository,
+        constraint = controller.load()?.registered_credix_lp_depositories.contains(&credix_lp_depository.key()) @UxdError::InvalidDepository,
+    )]
+    pub controller: AccountLoader<'info, Controller>,
+
+    /// #3 The first known active mercurial vault depository
+    #[account(
+        mut,
+        seeds = [
+            MERCURIAL_VAULT_DEPOSITORY_NAMESPACE,
+            mercurial_vault_depository.load()?.mercurial_vault.key().as_ref(),
+            mercurial_vault_depository.load()?.collateral_mint.as_ref()
+        ],
+        bump = mercurial_vault_depository.load()?.bump,
+        has_one = controller @UxdError::InvalidController,
+    )]
+    pub mercurial_vault_depository: AccountLoader<'info, MercurialVaultDepository>,
+
+    /// #4 The first known active credix lp depository
+    #[account(
+        mut,
+        seeds = [
+            CREDIX_LP_DEPOSITORY_NAMESPACE,
+            credix_lp_depository.load()?.credix_global_market_state.key().as_ref(),
+            credix_lp_depository.load()?.collateral_mint.as_ref()
+        ],
+        bump = credix_lp_depository.load()?.bump,
+        has_one = controller @UxdError::InvalidController,
+    )]
+    pub credix_lp_depository: AccountLoader<'info, CredixLpDepository>,
+}
+
+pub(crate) fn handler(ctx: Context<ComputeDepositoriesTargets>) -> Result<()> {
+    let controller = &ctx.accounts.controller.load()?;
+    let mercurial_vault_depository = &mut ctx.accounts.mercurial_vault_depository.load_mut()?;
+    let credix_lp_depository = &mut ctx.accounts.credix_lp_depository.load_mut()?;
+
+    // ---------------------------------------------------------------------
+    // -- Phase 1
+    // -- Read the desired weights for each depository on chain
+    // -- And generate a raw_target estimations that we can refine later
+    // ---------------------------------------------------------------------
+
+    // We want to balance the supply on weighted portions of the circulating supply for now
+    // This could be based on dynamic on-chain account values and liquidity later
```

Comment:
nit: please remove "maybe do later" things. It blurs what you are really doing here, now